my_robot_description/launch/my_robot_rviz.launch

```diff
@@ -24,17 +27,31 @@
 
   <!--
     Reads the robot model description, finds all joints, and publishes joint values to all nonfixed joints using GUI sliders. 
-  -->
+  --> 
   <node name="joint_state_publisher" 
     		pkg="joint_state_publisher" 
-    		type="joint_state_publisher" />
+    		type="joint_state_publisher"
+			output="screen"
+			unless="$(arg use_gui)"	/> <!-- Ktysai last two lines necessary to solve NO GUI for joint_state_publisher when using RVIZ   -->
+  <!-- Ktysai node necessary to solve NO GUI for joint_state_publisher when using RVIZ   -->		
+  <node name="joint_state_publisher_gui" 
+    		pkg="joint_state_publisher_gui" 
+    		type="joint_state_publisher_gui"
+			output="screen"
+			if="$(arg use_gui)"	/>
+
 
   <!-- 
     Reads the current robot joint states and publishes the 3D poses of each robot link using the kinematics tree build from the URDF. The 3D pose of the robot is published as ROS tf (transform). 
   -->
   <node name="robot_state_publisher" 
     		pkg="robot_state_publisher" 
-    		type="state_publisher" />
+    		type="robot_state_publisher" />
+  <!-- KTYSAI	
+    put type="robot_state_publisher", in order to make the robot aprear corectly in RVIz and deal with this type of error message:
+	"ERROR: cannot launch node of type [robot_state_publisher/state_publisher]: Cannot locate node of type [state_publisher] in package [robot_state_publisher].
+	Make sure file exists in package path and permission is set to executable (chmod +x)"
+  -->
```

Comment:
No need this comment here, `state_publisher` is an old name that was changed a long time ago.

Reply:
in the launch file it was like that, unfortunately.